internal/providers/openai.go

```
@@ -0,0 +1,99 @@
+	httpReq, err := http.NewRequestWithContext(ctx, "POST", r.baseURL+"/v1/chat/completions", bytes.NewReader(buf))
+	if err != nil {
+		return Response{}, err
+	}
+	httpReq.Header.Set("Content-Type", "application/json")
+	httpReq.Header.Set("Authorization", "Bearer "+r.apiKey)
+
+	resp, err := r.client.Do(httpReq)
+	if err != nil {
+		return Response{}, fmt.Errorf("openai: %w", err)
+	}
+	defer resp.Body.Close()
```

golangci-lint (2.12.1)

[error] Error return value of `resp.Body.Close` is not checked

(errcheck)